src/data/dataloader.py

```diff
@@ -25,6 +25,10 @@ def __init__(self, dataset, batch_size=1, shuffle=False, sampler=None,
                          worker_init_fn=worker_init_fn)
 
         if dataset.type == 'train':
+            if self.drop_last and grad_accumulation_steps > len(self._index_sampler):
```

Comment:
I think you mean `drop_last is True` but `grad_accumulation_steps > len(self._index_sampler)`.

If `drop_last is True`, the logic  is to drop the last uncompleted batch. Besides, the dataset can be divided by `batch_size`, which is defined as `len(self._index_sampler)`. 
Since we consider `grad_accumulation_steps * batch_size` as a completed batch, if `grad_accumulation_steps > len(self._index_sampler)`, the dataset (`len(self._index_sampler) * batch_size`) should be considered as **the last uncompleted batch**. That's why we need to throw an error.

Reply:
No, I mean drop_last = False but grad_accumulation_steps > len(self._index_sampler).

For example, the grad_accumulation_steps would be set to be larger than the total number of batch accidentally. In this case, it should raise an error in this place.

Reply:
If `drop_last is False`, the logic is **not** to drop the last uncompleted batch. Therefore, the situation `grad_accumulation_steps > len(self._index_sampler)` can be considered as the whole dataset is an uncompleted batch.